lesson_shell_fasta.md

```diff
@@ -0,0 +1,233 @@
+
+## Learning Objectives
+
+* Download files form FTP repositories
+* Working with compressed files
+* Working with FASTA files
+
+## Lesson
+
+In this lesson we will learn how to download data in [fasta format](https://en.wikipedia.org/wiki/FASTA_format) form the web and how to manipulate them using basic shell commands.
+We will use as sample data set the human [non-coding RNAs](https://en.wikipedia.org/wiki/Non-coding_RNA) (ncRNA).
+
+## Exercises
+### 1 - Downloading data
+
+The data we need are on the [Ensembl](http://www.ensembl.org/index.html) FTP site, a repository of all publicly available files from the project.
+
+There is an [html interface](http://www.ensembl.org/info/data/ftp/index.html) of the last release where it should be easy to locate the link to the FASTA file of the ncRNA
+
+*or*
+ We need to navigate the repository .
+You can navigate the public [FTP](ftp://ftp.ensembl.org/pub/) repository to find the file with all the sequences of human ncRNA (a little help: follow  last\_release>FASTA>homo_sapiens>ncrna)   
+
+*in both cases*
+
+we need to copy the full address that will look like this one:
+
+```
+ftp://ftp.ensembl.org/pub/release-81/fasta/homo_sapiens/ncrna/Homo_sapiens.GRCh38.ncrna.fa.gz
+```
+
+We will use this address as an argument to the command [wget](http://www.gnu.org/software/wget/manual/wget.html#Overview) to retrieve the data. Type in your terminal the command line below to download the file on your local machine.
+
+```
+$ wget <web address of the fasta file>
+```
+
+### 2 - Looking at the data
+
+The file we just downloaded is compressed and in the [gzip](https://www.gnu.org/software/gzip/manual/gzip.html) format. To see how big it is  use the *-l* and *-h* option of *ls*:
+
+```
+$ ls -lh
+
+-rw-rw-r-- 1 usr usr 8,4M ago 13 15:30 Homo_sapiens.GRCh38.ncrna.fa.gz
+```
+
+We can either work with the gzipped file or unzip it before, however because data sets can be really big it is probably worth working with the gzipped file.
```

Comment:
We can either work with the gzipped file or unzip it.  However, because the datasets can be really big it is probably worth working with the gzipped file.